src/commands/nix/install.rs

```diff
@@ -0,0 +1,212 @@
+use crate::commands::nix::{
+    link_generator::{CachixLinkGenerator, FlakeLinkInfo},
+    NIX_CMD, PRIORITY_FLAG, PROFILE_INSTALL_ARGS,
+};
+use anyhow::{anyhow, bail, Result};
+use clap::Parser;
+use std::{
+    fmt::Debug,
+    io::{BufRead, BufReader},
+    process::{Command, Stdio},
+    str::SplitWhitespace,
+    sync::mpsc,
+    thread,
+};
+use tracing::info;
+
+const NIX_PKG_PRIORITY_MSG: &str = "The conflicting packages have a priority of";
+const NIXOS_PRIORITY_MSG: &str = "have the same priority";
+
+#[derive(Debug, Parser)]
+pub struct NixInstallCommand {
+    /// Toolchain or component
+    pub name: String,
+    pub verbose: Option<bool>,
+}
+
+pub fn nix_install(command: NixInstallCommand) -> Result<()> {
+    let mut success = false;
+    let (priority_err, all_errs, link) = if command.is_toolchain() {
+        info!(
+            "downloading and installing fuel {} toolchain, this may take a while...",
+            command.name
+        );
+        let link = command.toolchain_link()?;
+        let mut priority_err = Vec::new();
+        let mut all_errs = Vec::new();
+        filter_command(
+            link.clone(),
+            command.name.clone(),
+            &mut priority_err,
+            &mut all_errs,
+            true,
+        );
+
+        (priority_err.concat(), all_errs.concat(), link)
+    } else if command.is_component() {
+        info!(
+            "downloading and installing component {}, this may take a while...",
+            command.name
+        );
+        let link = command.component_link()?;
+        let mut priority_err = Vec::new();
+        let mut all_errs = Vec::new();
+        filter_command(
+            link.clone(),
+            command.name.clone(),
+            &mut priority_err,
+            &mut all_errs,
+            false,
+        );
+
+        (priority_err.concat(), all_errs.concat(), link)
+    } else {
+        bail!(
+            "available distrubuted components:\n  -fuel-core\n  -fuel-core-client\n  -fuel-indexer\n  -forc\n  -forc-client\n  -forc-doc\n  -forc-explore\n  -forc-fmt\n  -forc-index\n  -forc-lsp\n  -forc-tx\n  -forc-wallet\n  -sway-vim\n
```

Comment:
Can we pull in this list dynamically rather than hardcoding it?

Reply:
Something like this?
```rust
const AVAILABLE_DIST_COMPONENTS: &[FuelComponent; 13] = &[
    FuelComponent::SwayVim,
    // ...
];
impl FuelComponent {
    fn as_str(&self) -> &'static str {
        match self {
            FuelComponent::SwayVim => "-sway-vim",
            // ...
        }
    }
}
// ...
let err_str = AVAILABLE_DIST_COMPONENTS.iter().map(|comp| comp.as_str()).collect::<Vec<&str>>().join('\n');
bail!("available distrubuted components: {err_str}\n")
```

Reply:
I've gone ahead and implemented it this way, lmk if you meant some other way and I can change it.